genCode.py

```diff
@@ -0,0 +1,169 @@
+        for module in modules:
+            print("from taskcluster.sync.{module} import {module}".format(module=module),
+                  file=fh)
+        print("__all__ = [", file=fh)
+        for module in modules:
+            print("    {module},".format(module=module), file=fh)
+        print("]", file=fh)
+
+
+def createRoutes(api):
+    """Create a string to define self.routes.
+    """
+    routes = ''
+    for entry in api['entries']:
+        if entry['type'] == 'function':
+            routes += "'%s': '%s',\n" % (entry['name'], anglesToBraces(entry['route']))
+    return routes
+
+
+def createRoutingKeys(api):
+    """Create a string to define self.routingKey
+
+    This doesn't include everything in routingKey, because
+        a) only a few things are currently used, and
+        b) for the sake of line and file length
+    """
+    parts = []
+    for entry in api['entries']:
+        if 'routingKey' in entry:
+            parts.append("    '%s': [" % entry['name'])
+            for routingKey in entry['routingKey']:
+                parts.append("        {")
+                for key in ROUTING_KEY_WHITELIST:
+                    if key in routingKey:
+                        value = routingKey[key]
+                        if isinstance(value, (six.binary_type, six.text_type)):
+                            value = "'{}'".format(value)
+                        parts.append("            '%s': %s," % (key, value))
+                parts.append("        },")
+            parts.append("    ],")
+    return '\n'.join(parts)
+
+
+def methodArgumentString(entry):
+    """Returns an argument string for the given function
+    """
+    parts = ['self']
+    parts.extend(argumentNames(entry))
+    if entry.get('query'):
+        parts.append('options=None')
+    return ", ".join(parts)
+
+
+def argumentNames(entry):
+    parts = []
+    if 'args' in entry:
+        parts.extend(entry['args'])
+    if 'input' in entry:
+        inputName = 'payload'
+        parts.append(inputName)
+    return parts
+
+
+def argumentString(entry):
+    """Returns a string of a list of argument names, sans self
+    """
+    string = ''
+    for name in argumentNames(entry):
+        string += "'{name}', ".format(name=name)
+    return string
+
+
+def anglesToBraces(s):
+    '''
+    Returns a string with <vars> replaced by {vars}
+    '''
+    if s:
+        return re.sub('<(.*?)>', '{\\1}', s)
+
+
+def render(env, templateName, serviceName, defn):
+    template = env.get_template(templateName)
+    api = defn['reference']
+    url = defn['referenceUrl']
+    return template.render(
+        serviceName=serviceName,
+        api=api,
+        methodArgumentString=methodArgumentString,
+        argumentString=argumentString,
+        createRoutes=createRoutes,
+        createRoutingKeys=createRoutingKeys,
+        generatedString=GENERATED_STRING,
+        referenceUrl=url,
+    )
+
+
+def docstringify(s, level=4):
```

Comment:
Can we use a more descriptive name, which doesn't use the ..ify suffix?  `typesetDocstring` would work?